src/rest/zone_client.erl

```diff
@@ -60,118 +61,179 @@ index_ready(Zone) when is_binary(Zone) ->
+get_dependent_zones(Zone, ToExclude) ->
     {error, not_implemented}.
 
 -spec is_instrumentation_enabled(Zone:: binary()) -> {ok, boolean()} | {error, any()}.
-is_instrumentation_enabled(_Zone) ->
+is_instrumentation_enabled(Zone) ->
     {error, not_implemented}.
 
 -spec set_instrumentation_enabled(Zone:: binary(), Enabled:: boolean()) -> ok | {error, any()}.
-set_instrumentation_enabled(_Zone, _Enabled) ->
+set_instrumentation_enabled(Zone, Enabled) ->
     {error, not_implemented}.
 
 -spec set_indexation_enabled(Zone:: binary(), Enabled:: boolean()) -> ok | {error, any()}.
-set_indexation_enabled(_Zone, _Enabled) ->
+set_indexation_enabled(Zone, Enabled) ->
     {error, not_implemented}.
 
 -spec simulate_load(Zone:: binary(), LoadRequest:: map()) -> ok | {error, any()}.
-simulate_load(_Zone, _LoadRequest) ->
+simulate_load(Zone, LoadRequest) ->
     {error, not_implemented}.
 
 %% function that blocks process until zone (all zones) returns 'true' from 'IP/index_ready' endpoint
 %%  - delay between consecutive requests may be constant, or Exponential Backoff can be used, whatever
 -spec wait_for_index(Zones:: binary() | list(binary()), Timeout:: integer()) -> ok | {error, any()}.
-wait_for_index(Zone, _Timeout) when is_binary(Zone) ->
+wait_for_index(Zone, Timeout) when is_binary(Zone) ->
     {error, not_implemented};
 
-wait_for_index(Zones, _Timeout) when is_list(Zones) ->
+wait_for_index(Zones, Timeout) when is_list(Zones) ->
     {error, not_implemented}.
 
--spec reaches(Algo:: naive | reaches, Zone:: binary(), From:: binary(), To:: binary()
+% MUST POTENTIAL
+-spec reaches(Algo:: naive | indexed, Zone:: binary(), From:: binary(), To:: binary()
     ) -> {ok, map()} | {error, any()}.
-reaches(_Algo, _Zone, _From, _To) ->
-    {error, not_implemented}.
-
--spec members(Algo:: naive | reaches, Zone:: binary(), Of:: binary()) -> {ok, map()} | {error, any()}.
-members(_Algo, _Zone, _Of) ->
-    {error, not_implemented}.
+reaches(Algo, Zone, From, To) ->
+    Path = case Algo of
+             naive -> <<"naive/reaches">>;
+             indexed -> <<"indexed/reaches">>
+           end,
+    {ok, Address} = http_utils:get_address(Zone),
+    Url = http_utils:build_url(Address, Path, [{<<"from">>, From}, {<<"to">>, To}]),
+    http_executor:post_request(Url, true).
+
+
+% MUST POTENTIAL
+-spec members(Algo:: naive | indexed, Zone:: binary(), Of:: binary()) -> {ok, map()} | {error, any()}.
+members(Algo, Zone, Of) ->
+    Path = case Algo of
+            naive -> <<"naive/members">>;
+            indexed -> <<"indexed/members">>
+          end,
+    {ok, Address} = http_utils:get_address(Zone),
+    Url = http_utils:build_url(Address, Path, [{<<"of">>, Of}]),
+    http_executor:post_request(Url, true).
 
 -spec effective_permissions(Algo:: naive | reaches, Zone:: binary(), From:: binary(),
     To:: binary()) -> {ok, map()} | {error, any()}.
-effective_permissions(_Algo, _Zone, _From, _To) ->
+effective_permissions(Algo, Zone, From, To) ->
```

Comment:
Dopisz tę funkcję. Ona wygląda naprawdę dokładnie tak samo jak te dwie wyżej, więc to żaden problem